src/main/java/com/google/api/codegen/transformer/ServiceTransformer.java

```diff
@@ -34,7 +36,20 @@ public ServiceDocView generateServiceDoc(
       docLines.add("");
       docLines.addAll(namer.getDocLines(conf.getManualDoc()));
     }
-    serviceDoc.lines(docLines.build());
+    List<String> lines = docLines.build();
+    serviceDoc.lines(lines);
+    if (lines != null) {
+      if (lines.size() > 0) {
+        serviceDoc.firstLine(lines.get(0));
+      } else {
+        serviceDoc.firstLine("");
+      }
+      if (lines.size() > 1) {
+        serviceDoc.remainingLines(lines.subList(1, lines.size()));
+      } else {
+        serviceDoc.remainingLines(new ArrayList<String>());
+      }
```

Comment:
```java
if (lines.isEmpty()) {
  serviceDoc.firstLine("");
  serviceDoc.remainingLines(Collections.emptyList());
} else {
  serviceDoc.firstLine(lines.get(0));

  // I think this works even if the size is 1. I also think we should copy the list, since sublist is an "image" of the original: changes in one is visible in another.
  serviceDoc.remainingLines(ImmutableList.copyOf(lines.subList(1, lines.size()));
}
```
?

Reply:
You are right - updated to something very similar.

I think that we don't need to copy - we are working with `ImmutableList` objects, so we don't need to worry about potential changes. Is that right, or am I not understanding the implications of sublist correctly?

Reply:
Oh yeah you're right.